mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -669,6 +603,89 @@ class PreferencesActivity : AbstractBaseActivity() {
         }
     }
 
+    internal class SendDeviceInfoSettingsFragment : AbstractSettingsFragment() {
+        override val titleResId: Int @StringRes get() = R.string.send_device_info_to_server_short
+        private lateinit var phoneStatePref: ItemUpdatingPreference
+        private lateinit var wifiSsidPref: ItemUpdatingPreference
+
+        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+            addPreferencesFromResource(R.xml.preferences_device_information)
+
+            val prefixPref = getPreference(PrefKeys.SEND_DEVICE_INFO_PREFIX)
+            val schedulePref = getPreference(PrefKeys.SEND_DEVICE_INFO_SCHEDULE)
+            phoneStatePref = getPreference(PrefKeys.SEND_PHONE_STATE) as ItemUpdatingPreference
```

Comment:
```suggestion
            phoneStatePref = getPreference(PrefKeys.SEND_PHONE_STATE)
```
This cast shouldn't be required.

Reply:
No, it is required. `getPreference()` returns `Preference`.